core/src/main/java/org/elasticsearch/action/bulk/BulkRequestHandler.java

```diff
@@ -71,9 +83,10 @@ public void execute(BulkRequest bulkRequest, long executionId) {
             try {
                 listener.beforeBulk(executionId, bulkRequest);
                 BulkResponse bulkResponse = Retry
-                        .on(EsRejectedExecutionException.class)
-                        .policy(backoffPolicy)
-                        .withSyncBackoff(client, bulkRequest);
+                    .on(EsRejectedExecutionException.class)
```

Comment:
Requiring a `ThreadPool` is the way forward that we chose after a few attempts. We will require a `ThreadPool` in the high level client too. It is simple enough to create. We will see what else can be done to make this better. The bulk processor is decoupled meaning that it doesn't require a Client anymore.